test/extensions/filters/http/cache/http_cache_implementation_test_common.cc

```diff
@@ -0,0 +1,749 @@
+#include "test/extensions/filters/http/cache/http_cache_implementation_test_common.h"
+
+#include <string>
+#include <utility>
+
+#include "envoy/extensions/cache/simple_http_cache/v3/config.pb.h"
+
+#include "source/common/common/assert.h"
+#include "source/extensions/filters/http/cache/cache_headers_utils.h"
+#include "source/extensions/filters/http/cache/http_cache.h"
+
+#include "test/mocks/http/mocks.h"
+#include "test/test_common/simulated_time_system.h"
+#include "test/test_common/utility.h"
+
+#include "absl/status/status.h"
+#include "absl/synchronization/notification.h"
+#include "gtest/gtest.h"
+
+using ::envoy::extensions::filters::http::cache::v3::CacheConfig;
+using ::testing::_;
+using ::testing::AnyNumber;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+namespace {
+
+CacheConfig getConfig() {
+  // Allows 'accept' to be varied in the tests.
+  CacheConfig config;
+  config.add_allowed_vary_headers()->set_exact("accept");
+
+  return config;
+}
+
+MATCHER(IsOk, "") { return arg.ok(); }
+
+} // namespace
+
+HttpCacheImplementationTest::HttpCacheImplementationTest()
+    : delegate_(GetParam()()), vary_allow_list_(getConfig().allowed_vary_headers()) {}
+
+void HttpCacheImplementationTest::SetUp() {
```

Comment:
any particular reason not to move the body of this into the ctor? Same with TearDown -> dtor below.